src/main/java/microsoft/exchange/webservices/data/util/DateTimeUtils.java

```diff
@@ -114,4 +115,577 @@ private static Date parseInternal(String value, boolean dateOnly) {
     };
   }
 
+  /**
+   * Convert Olson TimeZone to Microsoft TimeZone
+   * Generated using Unicode CLDR project
+   * Example: https://gist.github.com/scottmac/655675e9b4d4913c539c
+   */
+  public static String getMicrosoftTimeZoneName(TimeZone timeZone) {
+    String timeZoneId = timeZone.getID();
+    if(timeZoneId.equals("Africa/Abidjan")) return "Greenwich Standard Time";
```

Comment:
I think it will be better to use some Map implementation to store/search this information.